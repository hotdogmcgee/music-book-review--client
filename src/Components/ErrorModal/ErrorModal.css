.error-modal {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 20px;
    height: auto
}