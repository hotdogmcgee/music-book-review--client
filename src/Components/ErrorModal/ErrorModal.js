import React from 'react'
import './ErrorModal.css'
import { Button } from '../Utils/Utils'

export default class ErrorModal extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
      };
    }
  
  
    render() {
      const { show, handleClose, error } = this.props;
      const showHideClassName = show
        ? "modal display-block"
        : "modal display-none";
      return (
        <div className={showHideClassName}>
          <section className="modal-main error-modal">
            <div className="error-container">              <p>{error}</p></div>

            <Button onClick={handleClose}>Close</Button>
          </section>
        </div>
      );
    }
  }