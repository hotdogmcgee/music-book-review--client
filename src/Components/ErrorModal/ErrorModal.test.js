import React from "react";
import ErrorModal from "./ErrorModal";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("ErrorModal component", () => {
  it("renders a ErrorModal by default", () => {
    const wrapper = shallow(<ErrorModal />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the ErrorModal given props", () => {
    const wrapper = shallow(<ErrorModal message={"there was an error"} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
