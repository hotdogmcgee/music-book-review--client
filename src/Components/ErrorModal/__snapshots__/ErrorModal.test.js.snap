// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal component renders a ErrorModal by default 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main error-modal"
  >
    <p />
    <button>
      Close
    </button>
  </section>
</div>
`;

exports[`ErrorModal component renders the ErrorModal given props 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main error-modal"
  >
    <p />
    <button>
      Close
    </button>
  </section>
</div>
`;
